### ISSUE TITLE
rdkit.Chem.rdchem.KekulizeException: Can't kekulize mol. 

### ISSUE THREAD

**Issue by zorrosword1** [luancarvalhomartins/PyAutoFEP#55]
Our input molecules including **imidazole** fragments,. It seems like rdkit can't kekulize mol at Superimosed poses step. Is there any way to fix it? 

**================= Input ligands ==================
   Name         Molecule            Topology      
    A      [H]c1c([H])c(C(F)(F)F)c([H])c([H])c1N([H])c1c([H])c([H])c(S(=O)(=O)N([H])C([H])([H])[H])c([H])c1-c1nc([H])n(C([H])([H])[H])c1[H]   lig_data/A.itp   
    B      [H]c1c([H])c(C(F)(F)F)c([H])c([H])c1Oc1c([H])c([H])c(S(=O)(=O)N([H])C([H])([H])[H])c([H])c1-c1nc([H])n(C([H])([H])[H])c1[H]   lig_data/B.itp   
    C      [H]c1nc(N([H])c2c([H])c([H])c(S(=O)(=O)N([H])C([H])([H])[H])c([H])c2-c2nc([H])n(C([H])([H])[H])c2[H])c([H])c([H])c1C(F)(F)F   lig_data/C.itp   
==================================================**
=================== Poses read ===================
Name            File                 Details             
A               {'molecule': <rdkit.Chem.PropertyMol.PropertyMol object at 0x7f31f741dbf0>, 'topology': ['lig_data/A.itp']}                   
B               {'molecule': <rdkit.Chem.PropertyMol.PropertyMol object at 0x7f323d985df0>, 'topology': ['lig_data/B.itp']}                   
C               {'molecule': <rdkit.Chem.PropertyMol.PropertyMol object at 0x7f323e5e53b0>, 'topology': ['lig_data/C.itp']}                   
=============== Superimposed poses ===============
Name            File                      Note           
Traceback (most recent call last):
  File "../prepare_dual_topology.py", line 3451, in <module>
    verbosity=arguments.verbose, **arguments.poses_advanced_options)
  File "../prepare_dual_topology.py", line 2878, in align_ligands
    save_state=save_state, verbosity=verbosity, **kwargs)
  File "/home/zorrosword/PyAutoFEP/docking_readers/superimpose_loader.py", line 127, in superimpose_poses
    atom_map=this_atommap, **kwargs)
  File "/home/zorrosword/PyAutoFEP/os_util.py", line 517, in wrap_trace
    return f(*args, **kwargs)
  File "/home/zorrosword/PyAutoFEP/merge_topologies.py", line 243, in constrained_embed_shapeselect
    core_mol = rdkit.Chem.RemoveHs(core_mol)
rdkit.Chem.rdchem.KekulizeException: Can't kekulize mol.  Unkekulized atoms: 20

we will appreciate your kindly advise.

Thanks,

**Comment by luancarvalhomartins** [comment]
Thanks for your message and for using PyAutoFEP. RDKit sometimes is picky about less common moieties. I suggest you, first, to use .mol as input (in case you are using another format). In my experience, RDKit reads .mol format far better than, eg, .mol2. This is also true for the reference pose for superimpose. Also, give the last commit a try. I relaxed the sanitization after calling RemoveHs., this should help.

I manage to use it to prepare perturbations using you SMILES. Briefly, I converted your SMILES to .pdbqt and docked then to a dummy receptor with qvina2 using openbabel. Then I converted the SMILES to .mol files and use them to prepare perturbations. Using `pose_loader=superimpose` also worked. See relevant config files attached.

```
# Convert SMILES to mol files
obabel input_ligands.smi -opdbqt --gen3d -O "%s.mol" --split

# Now prepare the pdbqt for docking
obabel input_ligands.smi -opdbqt --gen3d -O "%s.pdbqt" --split

# NOTE: MODEL and ENDMDL lines must be remove from the pdbqt files. See receptor files in the attached tgz.
parallel --jobs=1 --line-buffer 'qvina2.1 --config vina_input.txt --ligand {}' ::: {A,B,C}.pdbqt

# Run prepare_dual_topology.py
python3 prepare_dual_topology.py --config_file config.ini
```

[20220711_imidazole.tar.gz](https://github.com/luancarvalhomartins/PyAutoFEP/files/9087399/20220711_imidazole.tar.gz)
 
```
# Convert a PDBQT to .mol
obabel poses/A_out.pdbqt -Oposes/A_out.mol

# So that we can test superimpose loader
python3 prepare_dual_topology.py --config_file config_superimpose.ini
```

[20220711_imidazole_superimpose.tar.gz](https://github.com/luancarvalhomartins/PyAutoFEP/files/9087619/20220711_imidazole_superimpose.tar.gz)

**Comment by luancarvalhomartins** [comment]
In case 1b5e139129462234089c31f56c2c5f07b2dfaef5 does fix your issue, please, mark this issue as fixed or just por here. In case it does not, please, let me know se I can further work on it.

Thank you.

**Comment by zorrosword1** [comment]
Thanks for your advise. It dose work for the older versions of PyAutoFEP(downloaded this May) and Rdkit(2020.09.1). However, for the latest versions of PyAutoFEP(downloaded last week) and Rdkit(2022.03.4), it gives the same error. I am not sure if it is the compatibility issues. 

Thank you. 



> In case [1b5e139](https://github.com/luancarvalhomartins/PyAutoFEP/commit/1b5e139129462234089c31f56c2c5f07b2dfaef5) does fix your issue, please, mark this issue as fixed or just por here. In case it does not, please, let me know se I can further work on it.
> 
> Thank you.

**Comment by luancarvalhomartins** [comment]
I can confirm that RDKit 2022.03.1b1.339 fails. I am committing a fix tomorrow, it seems to be something trivial.

**Comment by jiaming28** [comment]
I still get this error, and why this issue is closed.

**Comment by luancarvalhomartins** [comment]
Dear @jiaming28,

The issue is closed because the original bug was fixed by 1b5e139 and 56c337a. As sanitization errors are totally molecule-dependent, it is better to report a new issue for your specific case (which you already did, #67)

**Comment by jiaming28** [comment]
Thanks for your reply. I have posted my error message and input molecules. , https://github.com/luancarvalhomartins/PyAutoFEP/issues/67)
